pkg/istio/rbac/v1/provider.go

```diff
@@ -96,15 +101,167 @@ func (p *v1) ConvertAthenzModelIntoIstioRbac(m athenz.Model) []model.Config {
 func (p *v1) GetCurrentIstioRbac(m athenz.Model, csc model.ConfigStoreCache) []model.Config {
 
-	sr, err := csc.List(model.ServiceRole.Type, m.Namespace)
+	sr, err := csc.List(collections.IstioRbacV1Alpha1Serviceroles.Resource().GroupVersionKind(), m.Namespace)
 	if err != nil {
 		log.Errorf("Error listing the ServiceRole resources in the namespace: %s", m.Namespace)
 	}
 
-	srb, err := csc.List(model.ServiceRoleBinding.Type, m.Namespace)
+	srb, err := csc.List(collections.IstioRbacV1Alpha1Servicerolebindings.Resource().GroupVersionKind(), m.Namespace)
 	if err != nil {
 		log.Errorf("Error listing the ServiceRoleBinding resources in the namespace: %s", m.Namespace)
 	}
 
 	return append(sr, srb...)
 }
+
+// ConvertAthenzModelIntoIstioAuthzPolicy converts the Athenz RBAC model into Istio Authorization V1Beta1 specific
+// RBAC custom resource (AuthorizationPolicy)
+func (p *v1) ConvertAthenzModelIntoIstioAuthzPolicy(athenzModel athenz.Model, namespace string, serviceName string, svcLabel string) model.Config {
+	// authz policy is created per service. each rule is created by each role, and form the rules under
+	// this authz policy.
+	var out model.Config
+	// form authorization config meta
+	// namespace: service's namespace
+	// name: service's name
+	schema := collections.IstioSecurityV1Beta1Authorizationpolicies
+	out.ConfigMeta = model.ConfigMeta{
+		Type:      schema.Resource().Kind(),
+		Group:     schema.Resource().Group(),
+		Version:   schema.Resource().Version(),
+		Namespace: namespace,
+		Name:      serviceName,
+	}
+
+	// matching label, same with the service label
+	spec := &v1beta1.AuthorizationPolicy{}
+
+	spec.Selector = &workloadv1beta1.WorkloadSelector{
+		MatchLabels: map[string]string{"svc": svcLabel},
+	}
+
+	// generating rules, iterate through assertions, find the one match with desired format.
+	var rules []*v1beta1.Rule
+	for role, assertions := range athenzModel.Rules {
+		for _, assert := range assertions {
+			// assert.Resource contains the svc information that needs to parse and match
+			svc, path, err := common.ParseAssertionResource(zms.DomainName(athenz.NamespaceToDomain(namespace)), assert)
+			if err != nil {
+				continue
+			}
+			// if svc match with current svc, process it and add it to the rules
+			// note that svc defined on athenz can be a regex, need to match the pattern
+			res, e := regexp.MatchString(svc, svcLabel)
+			if e != nil {
+				log.Errorln("error matching string: ", e.Error())
+				continue
+			}
+			if !res {
+				log.Errorf("athenz svc %s does not match with current svc %s", svc, svcLabel)
+				continue
+			}
+			rule := &v1beta1.Rule{}
+			// form rule.From, must initialize internal source here
+			from_principal := &v1beta1.Rule_From{
+				Source: &v1beta1.Source{},
+			}
+			from_requestPrincipal := &v1beta1.Rule_From{
+				Source: &v1beta1.Source{},
+			}
+			// role name should match zms resource name
+			for _, roleName := range athenzModel.Roles {
+				if roleName == role {
```

Comment:
Can't we skip this outer for loop and if check and just directly use the `for _, roleMember := range athenzModel.Members[roleName]` loop instead? We already have the role here so we could use `role` instead of `roleName` to do the lookup against the map?